packages/catalog-process/src/services/catalogService.ts

```diff
@@ -2954,6 +2961,47 @@ export function catalogServiceBuilder(
         throw eServiceTemplateWithoutPublishedVersion(templateId);
       }
 
+      const tenant = await retrieveTenant(
+        ctx.authData.organizationId,
+        readModelService
+      );
+
+      assertTenantKindExists(tenant);
+
+      const riskAnalysis: RiskAnalysis[] = template.riskAnalysis
+        .filter((r) =>
+          match(tenant.kind)
+            .with(tenantKind.PA, () => r.tenantKind === tenantKind.PA)
+            .with(
+              tenantKind.GSP,
+              tenantKind.PRIVATE,
+              tenantKind.SCP,
+              () =>
+                r.tenantKind === tenantKind.GSP ||
+                r.tenantKind === tenantKind.PRIVATE ||
+                r.tenantKind === tenantKind.SCP
+              /**
+               * For now, GSP, PRIVATE, and SCP tenants share the same risk analysis.
+               * This may change in the future.
+               */
+            )
+            .exhaustive()
+        )
+        .map((r) => ({
+          id: generateId(),
+          createdAt: r.createdAt,
+          name: r.name,
+          riskAnalysisForm: r.riskAnalysisForm,
+        }));
+
+      if (template.mode === eserviceMode.receive && riskAnalysis.length === 0) {
+        throw receiveTemplateMissingTenantKindRiskAnalysis(
+          template.id,
+          tenant.id,
+          tenant.kind
+        );
+      }
```

Comment:
This piece of code could be a separate method `retrieveRiskAnalisysByTenantKind`, also statement that check conditions maybe a separate validator, `assertContainsRiskAnalysisInReceiveMode`